lib/msal-common/src/auth/ScopeSet.ts

```diff
@@ -153,7 +153,15 @@ export class ScopeSet {
         if (!otherScopes) {
             throw ClientAuthError.createEmptyInputScopeSetError(otherScopes);
         }
-        return this.unionScopeSets(otherScopes).size < (this.scopes.size + otherScopes.getScopeCount());
+
+        const unionScopes = this.unionScopeSets(otherScopes);
+
+        // Do not allow offline_access to be the only intersecting scope
+        const sizeOtherScopes = otherScopes.containsScope(Constants.OFFLINE_ACCESS_SCOPE)? otherScopes.getScopeCount() - 1 : otherScopes.getScopeCount();
+        const sizeThisScopes = this.containsScope(Constants.OFFLINE_ACCESS_SCOPE)? this.getScopeCount() - 1: this.getScopeCount();
+        const sizeUnionScopes = unionScopes.has(Constants.OFFLINE_ACCESS_SCOPE)? unionScopes.size - 1: unionScopes.size;
+
+        return sizeUnionScopes < (sizeThisScopes + sizeOtherScopes);
```

Comment:
Dumb q: Why are checking count than content? What if count matches and content differs for the set? I may be missing some logic and will check. This is mostly a question I immediately had.

Reply:
`unionScopes` is a set of scope array 1 + scope array 2 containing only unique entries. If the size of this is less than the size of both arrays added together then you know there were shared scopes between the 2 individual arrays

Reply:
Size of both sets, not arrays. these are all unique values.

Reply:
okay, got it.